enhancements/update/cvo-techpreview-manifests.md

```diff
@@ -0,0 +1,139 @@
+
+### Goals
+
+1. CVO honors an "only create or update this manifest in clusters with tech preview enabled".
+
+### Non-Goals
+
+1. Bootstrap rendering for tech preview operators.
+2. 4.y upgrades.  TechPreviewNoUpgrade explicitly disallows upgrades from 4.y to 4.y+1.
+3. Cleanup of manifests created as TechPreviewNoUpgrade.  TechPreviewNoUpgrade explicitly disallows being unset and this
+    is enforced using validation on the apiserver.
+4. Allow manifests to be removed or un-reconciled if TechPreviewNoUpgrade is *not* set.  There are a few case (mostly
+    around CRDs), where this capability is useful, but it is not as common as needing to create something additional.
+
+## Proposal
+
+Add an annotation that can be set on CVO manifests called `config.openshift.io/feature-gate` with one legal/honored value
+of "TechPreviewNoUpgrade".
+If the value is not "TechPreviewNoUpgrade", the manifest is never created.
+Since most people add manifests in order to see them applied, they will notice if their manifest is never created and
+the choice of annotation allows for potential future usage in the CVO consistent with feature gate handling used by
+other operators.
+
+If a manifest sets `.metadata.annotations["config.openshift.io/feature-gate"]="TechPreviewNoUpgrade"`, the manifest will
+not be created unless `featuregates.config.openshift.io|.spec.featureSet="TechPreviewNoUpgrade"`.
+If `featuregates.config.openshift.io|.spec.featureSet="TechPreviewNoUpgrade"`, then the manifest is reconciled as normal
+for whatever stage the CVO is in.
+The `featuregates.config.openshift.io|.spec.featureSet="TechPreviewNoUpgrade"` can be included in the installation
+manifests provided by a customer (we do this some CI scenarios as well), but there is no expected CVO change for bootstrapping.
+If bootstrapping changes are required, it will be responsibility of the contributing team to manage the proper inclusion
+in the installer and other bootstrapping components.
+
+Because the cluster cannot upgrade to 4.y+1 and because the TechPreviewNoUpgrade flag cannot be unset, there is no concern
+about removing any manifests that were added for TechPreviewNoUpgrade. 
+
+### User Stories
+
+### API Extensions
+
+### Implementation Details/Notes/Constraints [optional]
+
+### Risks and Mitigations
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+1. The TechPreview CI jobs (already present), should run after this feature is implemented.
+2. In 4.10, the cluster-api clusteroperator should not be present in our "normal" CI flows. 
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+When set, `TechPreviewNoUpgrade` prevents al 4.y to 4.y+1 upgrades.
+
+### Version Skew Strategy
+
+Because upgrades are prevented when this feature is active, there is no supported skew.
```

Comment:
There can still be skew between 4.y.z -> 4.y.z', and the CVO does not block those on `TechPreviewNoUpgrade` today, because the current `TechPreviewNoUpgrade` block is based on `Upgradeable=False` in a ClusterOperator, and since the SCC kerfuffle in 4.3, that no longer blocks patch updates.  We could teach the CVO to block patch updates on this property, or we could link [the `release.openshift.io/delete` enhancement][1] and point out that component maintainers have the same skew exposure as everyone else for patch updates.

[1]: https://github.com/openshift/enhancements/blob/ce4d303db807622687159eb9d3248285a003fabb/enhancements/update/object-removal-manifest-annotation.md